File: java/src/main/java/com/alxarafe/app/domain/catalogue/valueobject/Uom.java
package com.alxarafe.app.domain.catalogue.valueobject;

import java.util.Objects;

/**
 * Value Object representing a Unit of Measure.
 */
public record Uom(UomId id, UomCode code, String description) {

    public Uom {
        Objects.requireNonNull(id, "Uom id cannot be null.");
        Objects.requireNonNull(code, "Uom code cannot be null.");
        Objects.requireNonNull(description, "Uom description cannot be null.");
    }
}

File: php/src/Domain/Catalogue/ValueObject/Uom.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Domain\Catalogue\ValueObject;

/**
 * Value Object representing a Unit of Measure.
 */
final readonly class Uom
{
    public function __construct(
        private UomId $id,
        private UomCode $code,
        private string $description,
    ) {
    }

    public function id(): UomId
    {
        return $this->id;
    }

    public function code(): UomCode
    {
        return $this->code;
    }

    public function description(): string
    {
        return $this->description;
    }

    public function equals(self $other): bool
    {
        return $this->id->equals($other->id);
    }
}
